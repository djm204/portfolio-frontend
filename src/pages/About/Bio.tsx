import React, { ReactNode } from 'react'

type BioProps = {
  children: ReactNode
}

const Bio = (props: BioProps) => {
  return (
    <div className="WriteUp">
      <h2>About David</h2>
      {props.children}
    </div>
  )
}

export default Bio
