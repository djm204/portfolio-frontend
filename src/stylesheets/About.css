.About {
  padding-top: 100px;
}

.Bio {
  justify-content: center;
  align-content: space-between;
  padding-top: 50px;
}

.Bio__Image {
  text-align: center;
  flex-basis: 50%;
  font-size: 1.12em;
  padding-right: 57px;
}

.Bio__Image h4 .Bio__Image h5 {
  font-size: 1.5em;
}

.Bio__Image .Bio__Image__Picture img {
  border-radius: 30em;
  width: 300px;
  margin-bottom: 20px;
}

.WriteUp {
  letter-spacing: 0.09em;
  font-size: 1.12em;
}

.Quote {
  margin-top: 50px;
  border-top-left-radius: 10px;
  border-left: 5px solid rgba(172, 197, 255, 0.4);
  padding-left: 10px;
  font-style: italic;
  font-size: 1.05em;
  letter-spacing: 0.05em;
  padding: 5px 0px 5px 10px;
}
