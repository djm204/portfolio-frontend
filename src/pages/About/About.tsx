import React, { FunctionComponent } from 'react';

import { InfoSection } from '../../components/layout/InfoSection';
import Image from '../../components/Image';
import Bio from './Bio';
import '../../stylesheets/About.css';

const Home: FunctionComponent = () => {
    const bioPicUrl = 'david_on_a_mountain.jpg';

    return (
        <div
            id="About" 
            className="About"
            data-aos="fade-in"
            data-aos-easing="ease-in-out"
            data-aos-once="true"
            data-aos-duration="1000">
            <InfoSection classnames="Bio">
                <div className="Bio__Image">
                    <Image
                        imgClasses="Bio__Image__Picture"
                        imgUrl={bioPicUrl || ''}
                        altText="David Mendez" />
                    <h4>David Mendez</h4>
                    <h5>Developer</h5>
                </div>
                <div className="Bio__AndQuote">
                    <Bio>
                        <p>David is a life-long techy with an insatiable appetite for software development.</p>
                        <p>He spends much of his free time in front of the PC creating and furthering his knowledge about different technologies such as JavaScript, GoLang and Python.</p>
                        <p>Staying up to date with technology, code coverage and best practices are very important to David.</p>
                    </Bio>
                    <div className="Quote">
                        <h4>Favourite Quote</h4>
                        <blockquote>
                            "Why settle for normalcy when there's a whole world of weird to explore?"
                        </blockquote>
                    </div>
                </div>
            </InfoSection>
        </div>
    );
}

export default Home;