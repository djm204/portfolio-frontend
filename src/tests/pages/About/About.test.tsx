import React from 'react'
import About from '../../../pages/About/About'
import { render } from '@testing-library/react'
import { HashRouter as Router } from 'react-router-dom'

it('About renders without crashing', () => {
  const result = render(
    <Router>
      <About />
    </Router>
  )
  expect(result).toMatchSnapshot()
})
