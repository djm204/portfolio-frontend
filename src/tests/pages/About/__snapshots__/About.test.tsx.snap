// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About renders without crashing 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="About"
        data-aos="fade-in"
        data-aos-duration="1000"
        data-aos-easing="ease-in-out"
        data-aos-once="true"
        id="About"
      >
        <div
          class="InfoSection Bio"
          classnames="Bio"
        >
          <div
            class="Bio__Image"
          >
            <div
              class="image-wrapper Bio__Image__Picture"
            >
              <img
                alt="David Mendez"
                src="david_on_a_mountain.jpg"
              />
            </div>
            <h4>
              David Mendez
            </h4>
            <h5>
              Developer
            </h5>
          </div>
          <div
            class="Bio__AndQuote"
          >
            <div
              class="WriteUp"
            >
              <h2>
                About David
              </h2>
              <p>
                As a life-long technology enthusiast with an insatiable appetite for software development I spend a lot of my free time in front of the PC; creating and furthering my knowledge about different technologies such as JavaScript, GoLang and Python.
              </p>
              <p>
                Staying up to date with technology, code coverage and best practices are very important to me.
              </p>
            </div>
            <div
              class="Quote"
            >
              <h4>
                Favourite Quote
              </h4>
              <blockquote>
                "Why settle for normalcy when there's a whole world of weird to explore?"
              </blockquote>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="About"
      data-aos="fade-in"
      data-aos-duration="1000"
      data-aos-easing="ease-in-out"
      data-aos-once="true"
      id="About"
    >
      <div
        class="InfoSection Bio"
        classnames="Bio"
      >
        <div
          class="Bio__Image"
        >
          <div
            class="image-wrapper Bio__Image__Picture"
          >
            <img
              alt="David Mendez"
              src="david_on_a_mountain.jpg"
            />
          </div>
          <h4>
            David Mendez
          </h4>
          <h5>
            Developer
          </h5>
        </div>
        <div
          class="Bio__AndQuote"
        >
          <div
            class="WriteUp"
          >
            <h2>
              About David
            </h2>
            <p>
              As a life-long technology enthusiast with an insatiable appetite for software development I spend a lot of my free time in front of the PC; creating and furthering my knowledge about different technologies such as JavaScript, GoLang and Python.
            </p>
            <p>
              Staying up to date with technology, code coverage and best practices are very important to me.
            </p>
          </div>
          <div
            class="Quote"
          >
            <h4>
              Favourite Quote
            </h4>
            <blockquote>
              "Why settle for normalcy when there's a whole world of weird to explore?"
            </blockquote>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
